### PR TITLE
Intly 4256

### PR DESCRIPTION
**Pull request by philbrookes**
## What
Delete RHMI CR from BYOC clusters, before deprovisioning it.

### CODE REVIEW

File: jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile [hunk 1]
```diff
@@ -23,7 +23,48 @@ node("cirhos_rhel7") {
                     sh "ocm login --url=https://api.stage.openshift.com/ --token=${OCM_ACCESS_TOKEN}"
                 }
             } // stage
-            
+
+            stage("login to oc on BYOC clusters") {
+                script {
+                    if(params.BYOC == "true") {
+                        adminPassword = sh (
```
Comment (pawelpaszki) [inline jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile]: I think you need to declare this field, e.g. Prepend it with `String `

File: jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile [hunk 2]
```diff
@@ -23,7 +23,48 @@ node("cirhos_rhel7") {
...
+                            script: "cat ${installConfigDirectory}${clusterCredentialsFileName} | jq -r .admin.password",
+                            returnStdout: true
+                        ).trim()
+                        clusterId = sh (
```
Comment (pawelpaszki) [inline jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile]: there is `clusterID` declared above. You could reuse it here, e.g. replace with `clusterID`

File: jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile [hunk 3]
```diff
@@ -23,7 +23,48 @@ node("cirhos_rhel7") {
...
+                            script: "cat ${installConfigDirectory}${clusterCredentialsFileName} | jq -r .id",
+                            returnStdout: true
+                        ).trim()
+                        clusterAPI = sh (
+                            script: "ocm get cluster ${clusterId} | jq -r  .api.url",
```
Comment (pawelpaszki) [inline jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile]: replace with `clusterID`

File: jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile [hunk 4]
```diff
@@ -23,7 +23,48 @@ node("cirhos_rhel7") {
...
+                    if(params.BYOC == "true") {
+                        adminPassword = sh (
+                            script: "cat ${installConfigDirectory}${clusterCredentialsFileName} | jq -r .admin.password",
+                            returnStdout: true
+                        ).trim()
+                        clusterId = sh (
+                            script: "cat ${installConfigDirectory}${clusterCredentialsFileName} | jq -r .id",
+                            returnStdout: true
+                        ).trim()
+                        clusterAPI = sh (
```
Comment (pawelpaszki) [inline jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile]: I think you need to declare this field, e.g. Prepend it with `String `

File: jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile [hunk 5]
```diff
@@ -23,7 +23,48 @@ node("cirhos_rhel7") {
...
+                    if(params.BYOC == "true") {
+                        sh "oc delete rhmis -n ${params.NAMESPACE_PREFIX}operator"
+                    }
+                }
+            }
+
+            stage("Wait for RHMI CR to be removed on BYOC cluster") {
+                script {
+                    if(params.BYOC == "true") {
+                        boolean rhmiDeleted = isRHMIDeleted()
```
Comment (pawelpaszki) [inline jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile]: I think boolean has to start with capital letter

File: jobs/integr8ly/ocp4/osd/cluster/deprovision/osd-cluster-deprovision.yaml [hunk 6]
```diff
@@ -19,6 +19,14 @@
           name: 'RECIPIENTS'
           default: ''
           description: "Whitespace- or comma-separated list of email addresses to which the result of this pipeline execution will be sent"
+      - string:
+          name: "BYOC"
+          default: "false"
+          description: "Whether this cluster was provisioned using BYOC or not"
```
Comment (pawelpaszki) [inline jobs/integr8ly/ocp4/osd/cluster/deprovision/osd-cluster-deprovision.yaml]: this should be 
```
      - bool:
          name: 'BYOC'
          default: false
          description: "Indicates whether the cluster should have byoc features enabled"
```

File: jobs/integr8ly/ocp4/nightly/executor/Jenkinsfile [hunk 7]
```diff
@@ -167,7 +167,9 @@ def deprovisionCluster(clusterName) {
     if (buildStatus == 'SUCCESS') {
         parameters = [
             string(name: 'CLUSTER_NAME', value: clusterName),
-            string(name: 'RECIPIENTS', value: "${RECIPIENTS}")
+            string(name: 'RECIPIENTS', value: "${RECIPIENTS}"),
+            string(name: "BYOC", value: "${BYOC}"),
```
Comment (pawelpaszki) [inline jobs/integr8ly/ocp4/nightly/executor/Jenkinsfile]: replace with `booleanParam(name: 'BYOC', value:  Boolean.valueOf(BYOC)),`

File: jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile [hunk 8]
```diff
@@ -23,7 +23,48 @@ node("cirhos_rhel7") {
                     sh "ocm login --url=https://api.stage.openshift.com/ --token=${OCM_ACCESS_TOKEN}"
                 }
             } // stage
-            
+
+            stage("login to oc on BYOC clusters") {
+                script {
+                    if(params.BYOC == "true") {
```
Comment (pawelpaszki) [inline jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile]: replace `params.BYOC == "true"` with `params. BYOC.toString() == 'true'`

File: jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile [hunk 9]
```diff
@@ -23,7 +23,48 @@ node("cirhos_rhel7") {
...
+                            returnStdout: true
+                        ).trim()
+                        sh "oc login ${clusterAPI} --insecure-skip-tls-verify=true -u kubeadmin -p ${adminPassword}"
+                    }
+                }
+            }
+
+            stage("Remove RHMI CR on BYOC cluster") {
+                script {
+                    if(params.BYOC == "true") {
```
Comment (pawelpaszki) [inline jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile]: replace `params.BYOC == "true"` with `params. BYOC.toString() == 'true'`

File: jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile [hunk 10]
```diff
@@ -23,7 +23,48 @@ node("cirhos_rhel7") {
...
+                        sh "oc delete rhmis -n ${params.NAMESPACE_PREFIX}operator"
+                    }
+                }
+            }
+
+            stage("Wait for RHMI CR to be removed on BYOC cluster") {
+                script {
+                    if(params.BYOC == "true") {
```
Comment (pawelpaszki) [inline jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile]: replace `params.BYOC == "true"` with `params. BYOC.toString() == 'true'`

File: jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile (hunk 8 above)
Comment (philbrookes) [reply jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile]: I have changed this to `if(params.BYOC)` as it's a boolean, will that work?

Comment (pawelpaszki) [reply jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile]: it will

File: jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile (hunk 9 above)
Comment (pawelpaszki) [reply jobs/integr8ly/ocp4/osd/cluster/deprovision/Jenkinsfile]: or with `params.BYOC`

**Pull request merged**